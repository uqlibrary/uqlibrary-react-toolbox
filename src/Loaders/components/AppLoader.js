import React from 'react';
import PropTypes from 'prop-types';
import CircularProgress from 'material-ui/CircularProgress';

export default function AppLoader({title, logoImage, logoText, progressColor}) {
    return (
        <div className="app-loader columns is-gapless layout-fill" >
            <div className="app-loader-content column is-centered">
                <h1 className="display-2 color-reverse">{title}</h1>
                <br />
                <br />
                <CircularProgress size={80} thickness={8} color={progressColor} />
                <br />
                <br />
                {logoImage && <img src={logoImage} alt={logoText} />}
            </div>
        </div>
    );
}

AppLoader.propTypes = {
    title: PropTypes.string.isRequired,
    logoImage: PropTypes.string,
    logoText: PropTypes.string,
    progressColor: PropTypes.string
};

AppLoader.defaultProps = {
    progressColor: '#FFF'
};
