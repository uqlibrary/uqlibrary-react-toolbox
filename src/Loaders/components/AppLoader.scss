@import '../../../sass/variables';
@import '../../../sass/typography';

.app-loader {
  background-color: $theme-color;
  text-align: center !important;
  padding-top: calc(50% + 200px);

  .app-loader-content {
    position: absolute;
    height: 300px;
    border: 1px solid red;
    top: calc(50% - 150px);
    margin-left: auto;
    margin-right: auto;
  }
  > h1 {
    color: #FFF;
    text-align: center;
  }
}
