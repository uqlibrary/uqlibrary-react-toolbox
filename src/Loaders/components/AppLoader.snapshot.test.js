jest.dontMock('./AppLoader');

import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import AppLoader from './AppLoader';

function setup(title, logoImage, logoText, progressColor) {
    const props = {title, logoImage, logoText, progressColor};
    return shallow(<AppLoader {...props} />);
}

describe('AppLoader snapshots tests', () => {
    it('renders loader without image', () => {
        const wrapper = setup('Fez frontend');
        const tree = toJson(wrapper);
        expect(tree).toMatchSnapshot();
    });

    it('renders loader with image', () => {
        const wrapper = setup('Fez frontend', 'http://image/image.svg', 'Fez logo', '#CCC');
        const tree = toJson(wrapper);
        expect(tree).toMatchSnapshot();
    });
});