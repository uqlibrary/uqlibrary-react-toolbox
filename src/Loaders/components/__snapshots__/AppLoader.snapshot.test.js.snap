// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppLoader snapshots tests renders loader with image 1`] = `
<div
  className="app-loader"
>
  <div
    className="app-loader-content"
  >
    <h1
      className="display-2 color-reverse"
    >
      Fez frontend
    </h1>
    <br />
    <br />
    <CircularProgress
      color="#CCC"
      max={100}
      min={0}
      mode="indeterminate"
      size={80}
      thickness={8}
      value={0}
    />
    <br />
    <br />
    <img
      alt="Fez logo"
      src="http://image/image.svg"
    />
  </div>
</div>
`;

exports[`AppLoader snapshots tests renders loader without image 1`] = `
<div
  className="app-loader"
>
  <div
    className="app-loader-content"
  >
    <h1
      className="display-2 color-reverse"
    >
      Fez frontend
    </h1>
    <br />
    <br />
    <CircularProgress
      color="#FFF"
      max={100}
      min={0}
      mode="indeterminate"
      size={80}
      thickness={8}
      value={0}
    />
    <br />
    <br />
  </div>
</div>
`;
